run.py

```diff
@@ -803,8 +803,11 @@ def main():
         if dirpath == tdir:
             continue
 
-        # We only want to go one level deep.
-        dirnames[0:] = []
+        # Check if there are sub-test directories
+        if "test.yaml" in filenames or "check.sh" in filenames:
+            dirnames[0:] = []
```

Comment:
To me, this is python dark magic (`dirnames` value seems not used) but it was already like this

Reply:
Maybe we can add a comment that `dirnames` value reference gets used by `os.walk`

Reply:
OK. Thanks.